### PR TITLE
feat(packages): Add a wrapper setLocals function around native express.locals property

### PR DESCRIPTION
**Pull request by getspooky**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```
## What is the new behavior?
Adding setLocals function that persist a value throughout the life of the application.
example: 
```js
setLocals({title: 'My Nestjs App'});
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by kamilmysliwiec** [comment]
What's the reason for this change?

File: packages/core/adapters/http-adapter.ts [hunk 1]
```diff
@@ -86,6 +86,7 @@ export abstract class AbstractHttpAdapter<
   abstract initHttpServer(options: NestApplicationOptions);
   abstract useStaticAssets(...args: any[]);
   abstract setViewEngine(engine: string);
+  abstract setLocals(source: object);
```
Comment (Green-Cat) [inline packages/core/adapters/http-adapter.ts:89]: This will presumably now be missing in the fastify adapter.

**Comment by getspooky** [comment]
Hi @kamilmysliwiec 
Almost in all of my projects, I have a need to store some application based variables. In some cases, I change them on each request (for example –  counters). In some cases, they are not changed or changed infrequently. I want to be able to access them in almost any place of my application.

**Comment by kamilmysliwiec** [comment]
And why do we need a wrapper method for this? You can just call `getInstance().locals = ` instead.

**Pull request closed without merging**